server/routes/allocate-key-workers/controller.ts

```diff
@@ -0,0 +1,47 @@
+import { Request, Response } from 'express'
+import KeyworkerApiService from '../../services/keyworkerApi/keyworkerApiService'
+import { firstNameSpaceLastName } from '../../utils/formatUtils'
+import LocationsInsidePrisonApiService from '../../services/locationsInsidePrisonApi/locationsInsidePrisonApiService'
+
+export class AllocateKeyWorkerController {
+  constructor(
+    private readonly keyworkerApiService: KeyworkerApiService,
+    private readonly locationsApiService: LocationsInsidePrisonApiService,
+  ) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const prisonCode = res.locals.user.activeCaseLoad!.caseLoadId!
+    const records = await this.keyworkerApiService.searchPrisoners(req, prisonCode, {
+      query: req.query['query']?.toString() || '',
+      cellLocationPrefix: req.query['location']?.toString() || '',
+    })
+
+    const keyworkers = await this.keyworkerApiService.getKeyworkerMembers(req, prisonCode, { status: 'ALL' })
```

Comment:
The legacy code round this is misleading and it wasn't specified in the ticket. The correct status filter to use is 'ACTIVE' (see [allocateKeyWorker.js](https://github.com/ministryofjustice/manage-key-workers/blob/main/backend/controllers/allocateKeyWorker.js#L65)). The legacy code pulls back all the key workers in order to populate the name of each person's current keyworker. We don't need to do that as the search prisoners endpoint already populates that